### DISCUSSION TITLE
关于通过 freesql 使用 MySQL 的开源遵守疑问

### DISCUSSION THREAD

**Discussion by U11Leung** [dotnetcore/FreeSql#1124] [General]
可能是个幼稚的问题，但是想问，在写的闭源的商业软件，用它来连接 MySQL，会被 GPL 传染吗，freesql 用了 oracle 的 connector 之类的 GPL 组件吗？

**Comment by 2881099** [comment]
FreeSql.dll 是 MIT最宽容开源协议，并且无任何依赖。

FreeSql.Provider.MySql.dll MIT 引用 MySql.Data.dll GPL，前者任何是 MIT，修改后者需要开源。

另一个也可以访问 mysql 驱动 FreeSql.Provider.MySqlConnector.dll MIT 引用 MySqlCobnector.dll MIT，都是 MIT 没任何限制。

**Comment by 2881099** [comment]
oracle 因为只有官方驱动，所以遵从官方的开源协议就好。

GPL 可以商用的，只是修改源码也要开源。

**Comment by 2881099** [comment]
修改 mysql.data.dll ，oracle 等 adonet 驱动才会受限制，修改 freesql 任何代码都不受限制，MIT

**Comment by U11Leung** [comment]
所以，如果只是单纯通过引入 FreeSql.dll 或者 FreeSql.Provider.MySql.dll 或者 FreeSql.Provider.MySqlConnector.dll ，但不修改其他文件，去连接 MySQL 服务器，是不需要考虑遵守 GPL 的，只需要 MIT，我理解正确吗？

**Reply by luoyunchong** [reply]
使用二进制，不需要管开源协议！

**Reply by U11Leung** [reply]
> 使用二进制，不需要管开源协议！

那，也不需要包含 MIT 的文字在我的软件里了吗？

**Reply by 2881099** [reply]
不需要
